src/style/color.rs

```diff
@@ -141,14 +140,90 @@ impl Color {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
 #[cfg(feature = "serde")]
 impl<'de> serde::Deserialize<'de> for Color {
+    /// This is used to deserialize a value into Color via serde.
+    ///
+    /// This implementation uses the `FromStr` trait to deserialize strings, so named colours, RGB,
+    /// and indexed values are able to be deserialized. In addition, values that were produced by
+    /// the the older serialization implementation of Color are also able to be deserialized.
+    ///
+    /// See the [`Color`] documentation for more information on color names.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use ratatui::prelude::*;
+    ///
+    /// #[derive(serde::Deserialize)]
+    /// struct Theme {
+    ///     color: Color,
+    /// }
+    ///
+    /// let theme: Theme = serde_json::from_str::<Theme>("{\"color\": \"bright-white\"}").unwrap();
+    /// assert_eq!(theme.color, Color::White);
+    ///
+    /// let theme: Theme = serde_json::from_str::<Theme>("{\"color\": \"#00FF00\"}").unwrap();
+    /// assert_eq!(theme.color, Color::Rgb(0, 255, 0));
+    ///
+    /// let theme: Theme = serde_json::from_str::<Theme>("{\"color\": \"42\"}").unwrap();
+    /// assert_eq!(theme.color, Color::Indexed(42));
+    ///
+    /// let theme: Result<Theme, _> = serde_json::from_str::<Theme>("{\"color\": \"invalid\"}");
+    /// assert!(theme.is_err());
+    ///
+    /// // Deserializing from the previous serialization implementation
+    /// let theme: Theme = serde_json::from_str::<Theme>("{\"color\": {\"Rgb\":[255,0,255]}}").unwrap();
+    /// assert_eq!(theme.color, Color::Rgb(255, 0, 255));
+    ///
+    /// let theme: Theme = serde_json::from_str::<Theme>("{\"color\": {\"Indexed\":10}}").unwrap();
+    /// assert_eq!(theme.color, Color::Indexed(10));
+    /// ```
     fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
     where
         D: serde::Deserializer<'de>,
     {
-        let s = String::deserialize(deserializer)?;
-        FromStr::from_str(&s).map_err(serde::de::Error::custom)
+        /// Colors are currently serialized with the `Display` implementation, so
+        /// RGB values are serialized via hex, for example "#FFFFFF".
+        ///
+        /// Previously they were serialized using serde derive, which encoded
+        /// RGB values as a map, for example { "rgb": [255, 255, 255] }.
+        ///
+        /// The deserialization implementation utilises a `Helper` struct
+        /// to be able to support both formats for backwards compatibility.
+        #[derive(serde::Deserialize)]
+        enum ColorWrapper {
+            Rgb(u8, u8, u8),
+            Indexed(u8),
+        }
+
+        #[derive(serde::Deserialize)]
+        #[serde(untagged)]
+        enum Helper {
+            String(String),
+            ColorWrapper(ColorWrapper),
+        }
```

Comment:
> Would it be fine if the value being deserialized is not a string or one of the old formats (for example passing an empty array), it would create this error Failed to parse Colors: data did not match any variant of untagged enum Helper? Or should I rename Helper to Color?

If this ends up appearing in error messages, then it's worth thinking about how best to make the failure obvious to the user. "Helper" is purely related to the Color deserialization. Perhaps this should be ColorOrString or ColorVersionedSerialization, DeserializationFormat or something similar to those names (I'm sure you could think of a better name than those options). Perhaps even name the variants V1 / V2 or Version1 / Version2 to capture the intent rather than the implementation detail?